upcoming: [DI-22920] - Add megamenu changes

ankita-akamai opened the pull request:
## Description 📝
Mega Menu/Navigation changes for ACLP

## Changes  🔄
- Primary navigation is updated to have metrics and alerts as two different menu items instead of being shown together as two tabs under the main monitor tab.
- Separate landings for metrics and alerts with headers being displayed as 'Metrics' and 'Alerts' respectively.
- Move tabs(definitions, etc.) out of the paper component in the alerts.

## Target release date 🗓️
Next Release

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/user-attachments/assets/ddf75285-5509-422f-b3c7-8700b7fd75cb) | ![image](https://github.com/user-attachments/assets/470dcac5-14ed-4156-82e2-e2403e3ebdfa)|
| ![image](https://github.com/user-attachments/assets/51a94607-ca3a-4a38-85a1-cc3e520eb31a) | ![image](https://github.com/user-attachments/assets/682dd5ee-18a3-4304-9642-4aecbf581ae2)|
| ![image](https://github.com/user-attachments/assets/8b1c5cfe-46b6-4564-a213-071edfd42201) |![image](https://github.com/user-attachments/assets/ef87b34c-803d-43fb-9f76-c8abab018680)|
| ![image](https://github.com/user-attachments/assets/f1cfdb0e-85b6-4680-8d4e-367bf160dd68) |![image](https://github.com/user-attachments/assets/5bfabfe9-1f69-4491-a9b0-a7e6e18cf222)|

## How to test 🧪

### Verification steps

(How to verify changes)

- Navigate to monitor, see the menu items displayed. They should be these 3 in the order shown - 'Metrics', 'Alerts', 'Longview'.
- Verify that the previous 'Monitor' consisting of 'dashboards and 'alerts' tabs are now correctly split into two menu options - 'Metrics' and 'Alerts'.

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

</details>

---

ankita-akamai commented:
@agorthi-akamai can you correct the test statements stating 'monitor navigation item' ? We don't have monitor as navigation tab anymore. This is at various places in the cypress tests.

coliu-akamai reviewed (commented):
Looks like this PR is causing a lot of test failures with placement groups - I haven't looked too closely yet, but could they be related to changes with the primary nav / main content?

ankita-akamai commented:
> Looks like this PR is causing a lot of test failures with placement groups - I haven't looked too closely yet, but could they be related to changes with the primary nav / main content?

@coliu-akamai These cypress tests which are failing are unrelated to these changes.

ankita-akamai commented:
> Looks like this PR is causing a lot of test failures with placement groups - I haven't looked too closely yet, but could they be related to changes with the primary nav / main content?

I have fixed the cypress failures, looks like a wrapper in main-content file was somehow interfering with placement groups unexpectedly. Removed it and there are no more failures in placement groups. There is a failure in edit-user-alert.spec which is unrelated and a separate pr has been raised to fix this spec - https://github.com/linode/manager/pull/11822. Thanks for the heads up!

coliu-akamai commented on packages/manager/src/routes/index.tsx:
```diff
@@ -16,6 +16,7 @@ import { kubernetesRouteTree } from './kubernetes';
 import { linodesRouteTree } from './linodes';
 import { longviewRouteTree } from './longview';
 import { managedRouteTree } from './managed';
+import { CloudPulseMetricsRouteTree } from './metrics';
```
nit: we should uncapitalize the first letter of this route tree to follow convention with the other route trees

coliu-akamai reviewed (approved):
thanks @ankita-akamai, routing is working as expected. Approving pending one small comment

nikhagra-akamai commented:
merging the PR as it has required approval & all checks are passed
